.github/workflows/ci.yml

```
@@ -57,65 +60,67 @@ jobs:
-            # the parser does not like numbers with a leading zero.
-            $suffix = 'g' + $(git rev-parse --short HEAD)
-            $params = "--version-suffix", $suffix
-          }
+          name: consul-logs-${{ matrix.consul }}-${{ matrix.framework }}-${{ matrix.os }}
+          path: consul*.log
 
-          dotnet pack --configuration=Release --no-build @params Consul/Consul.csproj
+  package:
+    runs-on: windows-2019
+    needs: [Consul_AspNetCore, Consul]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Create NuGet package
+      run: |
+        if (-not ("${{ github.ref }}" -like "refs/tags/v*")) {
+          # WORKAROUND: Add letter prefix to ensure that MSBuild treats
+          # the suffix as a string. e.g. '0313071' will fail as an invalid
+          # version string, but 'G0313071' will succeeded. It looks like
+          # the parser does not like numbers with a leading zero.
+          $suffix = 'g' + $(git rev-parse --short HEAD)
+          $params = "--version-suffix", $suffix
+        }
+
+        dotnet pack --configuration=Release --output dist
 
-          if ("${{ github.ref }}" -like "refs/tags/v*") {
+        if ("${{ github.ref }}" -like "refs/tags/v*") {
             $tag = "${{ github.ref }}".SubString(11)
             $expectedFile = "dist/consul/Consul.$tag.nupkg"
             # Check whether the tag and the package version match together
             if (-not (Test-Path -Path $expectedFile)) {
                 echo "Expected file $expectedFile doesn't exist"
                 exit 1
             }
-          }
-
-      - name: Upload NuGet package artifact
-        if: runner.os == 'Windows'
-        uses: actions/upload-artifact@v2
-        with:
-          name: nuget-package
-          path: dist/consul
+        }
+    - name: Upload NuGet package artifact
+      uses: actions/upload-artifact@v2
+      with:
+        path: dist/*.nupkg
 
   # Publish NuGet package when a tag is pushed.
-  # Tests need to succeed on all platforms first, including having a tag name that matches the version number.
+  # Tests need to succeed for all components and on all platforms first,
+  # including having a tag name that matches the version number.
   publish:
-    if: startsWith(github.ref, 'refs/tags/v')
-    needs: build
+    if: github.ref == 'refs/heads/master' && startsWith(github.ref, 'refs/tags/v')
+    needs: package
     runs-on: ubuntu-18.04
     steps:
       - name: Download NuGet package artifact
         uses: actions/download-artifact@v2
         with:
-          name: nuget-package
-          path: dist/consul
+          path: dist
       - name: Publish to NuGet
-        run: dotnet nuget push dist/consul/*.nupkg --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json
+        run: dotnet nuget push dist/*.nupkg --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json
```

I think we cannot push `Consul.AspNetCore` package yet, So maybe let's use more specific path here:
```suggestion
        run: dotnet nuget push dist/Consul.1*.nupkg --api-key ${{ secrets.NUGET_API_KEY }} --source https://api.nuget.org/v3/index.json
```

---

Ok